src/services/expenses.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+let nextExpenseId = 1;
+
+let expenses = [];
+
+const getExpensesData = () => {
+  return expenses;
+};
+
+const clearExpensesArray = () => {
+  expenses = [];
+};
+
+function postExpense(body) {
+  const expense = {
+    id: nextExpenseId++,
+    ...body,
+  };
+
+  expenses.push(expense);
+
+  return expense;
+}
+
+function getExpenses(query) {
+  const { userId, category, from, to } = query;
+  const numberFromDate = new Date(from).getTime();
+  const numberToDate = new Date(to).getTime();
+  let copy = [ ...expenses ];
```

Comment:
You don't need this `copy`, because `filter` creates a new array.

Reply:
I use a copy because if I need to filter not only one user ID, but for example the user ID and category and some other name, that is, I need to filter the array three times.